collab2/foraging/toolkit/velocity.py

```diff
@@ -3,93 +3,51 @@
 import numpy as np
 import pandas as pd
 
-from collab.foraging.toolkit.trace import rewards_trace
-from collab.foraging.toolkit.utils import generate_grid
-from collab.foraging.toolkit.visibility import filter_by_visibility
 
-
-def add_velocities_to_foragers(foragers: List[pd.DataFrame]) -> None:
-
-    for forager in foragers:
-        forager["velocity_x"] = forager["x"].diff().fillna(0)
-        forager["velocity_y"] = forager["y"].diff().fillna(0)
-
-
-def add_velocities_to_data_object(data_object) -> None:
-    add_velocities_to_foragers(data_object.foragers)
-    data_object.foragersDF = pd.concat(data_object.foragers)
-
-
-def generate_velocity_scores(
-    sim,
-    velocity_time_decay=3,
-    velocity_spatial_decay=0.15,
-    time_shift=0,
-    grid=None,
-    visibility_restriction="visible",
+# a function that calculates the velocity magnitude and direction and outputs updated (foragers, foragersDF)
```

Comment:
I think these functions become somewhat specific and need docstrings, but their formatting should be different (for instance, to allow for automatic compilation by sphinx into a html documentation later on). Please visit 

https://basisresearch.github.io/chirho/explainable.html

for an example, and then explore the chirho codebase to see how the docstrings are formatted there.  Please copy their style.

Reply:
At least once we settle down on the final form of the function (backward/forward?)

Reply:
Added docstrings to the functions. We can polish them later if we want something more concise and formal for an html documentation.

Sending you a slack message to discuss the question of backward/forward velocity.